lib/ctcs2_to_junit.pm

```diff
@@ -100,20 +100,20 @@ sub generateXML {
         if ((get_var('XFSTESTS') || get_var('QA_TESTSET')) && ($case_status eq 'failure' || $case_status eq 'skipped')) {
             (my $test_path = $test) =~ s/-/\//;
             $test_path = '/opt/log/' . $test_path;
-            my $test_out_content = script_output("if [ -f $test_path ]; then cat $test_path| sed \"s/'//g\" | tr -cd '\\11\\12\\15\\40-\\176' | tail -n 200; else echo 'Test Crashed, find log in serial0.txt'; fi", 600);
+            my $test_out_content = script_output("if [ -f $test_path ]; then tail -n 200 $test_path | sed \"s/'//g\" | tr -cd '\\11\\12\\15\\40-\\176'; else echo 'Test Crashed, find log in serial0.txt'; fi", 600);
             $writer->startTag('system-out');
             $writer->characters($test_out_content);
             $writer->endTag('system-out');
             if ($case_status eq 'failure') {
                 my $test_err_content = script_output("
                     echo '====out.bad log====';
                     if [ -f $test_path.out.bad ];
-                        then cat $test_path.out.bad | sed \"s/'//g\" | tr -cd '\\11\\12\\15\\40-\\176' | tail -n 200;
+                        then tail -n 200 $test_path.out.bad | sed \"s/'//g\" | tr -cd '\\11\\12\\15\\40-\\176';
```

Comment:
What about to remove tail as well? You can place filename to sed command directly.

Reply:
The `tr` could be also included inside the `sed` as well.